cmd/fetch_policy.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//////////////////////////////////////////////////////////////////////////////
+//
+// Define the `ec fetch policy <source-url>` command.
+//
+// As it says in the docs it's not generally needed for the EC contract, but
+// could be useful generally for hacking and troubleshooting purposes.
+//
+// If/when we designate some commands as "experimental" or "plumbing" then this
+// could be one of them.
+//
+//
+package cmd
+
+import (
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/hacbs-contract/ec-cli/internal/downloader"
+	"github.com/hacbs-contract/ec-cli/internal/utils"
+)
+
+func fetchPolicyCmd() *cobra.Command {
+	var destDir string = "policy"
+	var useWorkDir bool = false
+
+	cmd := &cobra.Command{
+		Use:   "policy <source-url>",
+		Short: "Fetch policy rules from a git repository or other source",
+		Long: `Fetch policy rules (rego files) from a git repository or other source.
+
+This command is actually a thin wrapper for 'conftest pull' so you can refer to the
+conftest pull documentation for more details on the different types of supported source
+URLs.
+
+This is not typically required to verify the Enterprise Contract, but it's exposed for
+troubleshooting and debugging purposes.
+`,
+		Example: `Fetch rego files from a git repo:
+
+  ec fetch policy github.com/hacbs-contract/ec-policy//policy/lib --dest-dir policy/lib`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			sourceUrl := args[0]
+			fullDestDir := destDir
+
+			if useWorkDir {
+				workDir, err := utils.CreateWorkDir()
+				if err != nil {
+					log.Debug("Failed to create work dir!")
+					return err
+				}
+				fullDestDir = filepath.Join(workDir, destDir)
+			}
+
+			return downloader.Download(cmd.Context(), fullDestDir, sourceUrl)
```

Comment:
Maybe we need to write to stdout to tell the user where exactly it was downloaded, specially if `useWorkDir` is `true`.

Reply:
I was thinking about that, and I considered doing a log.Info(...) here, but I decided against it since you can do `--verbose` and get a similar message from the internal/downloader.

Reply:
On the other hand, you'd probably always want to see the random tmp work dir...

Reply:
Yeah, I think stdout is probably a good place for that info.

Reply:
As a lazy solution, I mentioned in the docs that you should use --verbose to see the work dir.

Reply:
I think putting it to stdout would be preferable as well.

Reply:
Done in the current revision.